thinc/backends/ops.py

```diff
@@ -300,29 +321,31 @@ def pad(  # noqa: F811
         # array sizes.
         length = (length + (round_to - 1)) // round_to * round_to
         final_shape = (len(seqs), length) + seqs[0].shape[1:]
-        output: Array3d = self.alloc(final_shape, dtype=seqs[0].dtype)
+        output: ArrayTXNotMind = self.alloc(final_shape, dtype=seqs[0].dtype)
         for i, arr in enumerate(seqs):
             # It's difficult to convince this that the dtypes will match.
             output[i, : arr.shape[0]] = arr  # type: ignore
         return output
 
-    def unpad(self, padded: Array3d, lengths: List[int]) -> List2d:
+    def unpad(
+        self, padded: ArrayTXNotMind, lengths: List[int]
+    ) -> List[ArrayTXNotMaxd_co]:
         """The reverse/backward operation of the `pad` function: transform an
         array back into a list of arrays, each with their original length.
         """
         output = []
         for i, length in enumerate(lengths):
             output.append(padded[i, :length])
-        return cast(List2d, output)
+        return cast(List[ArrayTXNotMaxd_co], output)
 
-    def list2padded(self, seqs: List[Floats2d]) -> Padded:
+    def list2padded(self, seqs: List[ArrayT2d_co]) -> Padded:
```

Comment:
I could see no reason not to support Ints here as well as Floats

Reply:
Agreed! But the current typing also supports a `List` mixing both, no? Is that what we want?

Reply:
We then revised this decision because such strict variables cannot be used as function call parameters.